src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,230 @@
+package duke;
+
+import duke.exceptions.EmptyListException;
+import duke.exceptions.TaskDoneException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Duke {
+
+    private static void readFile(List<Task> tasks) {
+        try {
+            FileInputStream file = new FileInputStream("src/main/java/duke/duke.txt");
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                String[] parts = line.split("\\|", 0);
+                String type = parts[0];
+                String isDone = parts[1];
+                String task = parts[2];
+                if (type.equals("T")) {
```

Comment:
maybe you should move "T" to a constant in order to avoid the need of magic strings.